<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>turnarond</author>
 <class>TabModbusCluster</class>
 <widget class="QWidget" name="TabModbusCluster">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus æ‰¹é‡åŒæ­¥æµ‹è¯•å·¥å…·</string>
  </property>
  <layout class="QHBoxLayout" name="mainLayout" stretch="3,1">
   <property name="spacing">
    <number>6</number>
   </property>
   <property name="leftMargin">
    <number>6</number>
   </property>
   <property name="topMargin">
    <number>6</number>
   </property>
   <property name="rightMargin">
    <number>6</number>
   </property>
   <property name="bottomMargin">
    <number>6</number>
   </property>
   <property name="layoutStretch" stdset="0">
    <string>2,1</string>
   </property>
   <item>
    <widget class="QTableWidget" name="table_registerMatrix">
     <property name="font">
      <font>
       <family>Courier New</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::EditTrigger::DoubleClicked|QAbstractItemView::EditTrigger::SelectedClicked</set>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectionBehavior::SelectItems</enum>
     </property>
     <property name="rowCount">
      <number>10</number>
     </property>
     <property name="columnCount">
      <number>4</number>
     </property>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <column>
      <property name="text">
       <string>è¯»å–æ—¶é—´</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>å¯„å­˜å™¨åœ°å€</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>å¾…å†™å€¼</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>è®¾å¤‡A</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="rightPanelLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox_config">
       <property name="title">
        <string>é…ç½®</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_config">
        <item row="0" column="0">
         <widget class="QLabel" name="label_port">
          <property name="text">
           <string>Modbus ç«¯å£ï¼š</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEdit_port">
          <property name="text">
           <string>502</string>
          </property>
          <property name="maxLength">
           <number>5</number>
          </property>
          <property name="placeholderText">
           <string>502</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_slaveId">
          <property name="text">
           <string>ä»ç«™ IDï¼š</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLineEdit" name="lineEdit_slaveId">
          <property name="text">
           <string>1</string>
          </property>
          <property name="maxLength">
           <number>3</number>
          </property>
          <property name="placeholderText">
           <string>1~247</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_startAddr">
          <property name="text">
           <string>èµ·å§‹åœ°å€ï¼š</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spin_startAddr">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_length">
          <property name="text">
           <string>é•¿åº¦ï¼ˆæ•°é‡ï¼‰ï¼š</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QSpinBox" name="spin_length">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>125</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="4">
         <layout class="QHBoxLayout" name="hLayout_regType">
          <item>
           <widget class="QLabel" name="label_regType">
            <property name="text">
             <string>å¯„å­˜å™¨ç±»å‹ï¼š</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_regType">
            <item>
             <property name="text">
              <string>ä¿æŒå¯„å­˜å™¨ (Holding Registers, 4x)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>çº¿åœˆ (Coils, 0x)</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_control">
       <property name="title">
        <string>æ“ä½œ</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_control">
        <item>
         <widget class="QPushButton" name="btn_readAll">
          <property name="styleSheet">
           <string notr="true">background-color: #2196F3; color: white;</string>
          </property>
          <property name="text">
           <string>â–¶ è¯»å–å¹¶å¯¹æ¯”</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="hLayout_refreshRow">
          <item>
           <widget class="QPushButton" name="btn_startAutoRefresh">
            <property name="text">
             <string>ğŸ” å®šæ—¶åˆ·æ–°ï¼ˆå…³é—­ï¼‰</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spin_refreshInterval">
            <property name="suffix">
             <string> ç§’</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>60</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_refresh">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="btn_writeSelected">
          <property name="styleSheet">
           <string notr="true">background-color: #FF9800; color: white;</string>
          </property>
          <property name="text">
           <string>ğŸ”¥ å†™å…¥é€‰å®šè®¾å¤‡</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_control">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>10</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_duration">
          <property name="text">
           <string>â±ï¸ è®¾å¤‡æ“ä½œè€—æ—¶:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_writeOptions">
       <property name="title">
        <string>å†™å…¥é€‰é¡¹</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_writeOptions">
        <item>
         <widget class="QRadioButton" name="radio_writeSingleValue">
          <property name="text">
           <string>å‘é€‰å®šè®¾å¤‡çš„æ‰€æœ‰å¯„å­˜å™¨å†™å…¥ç›¸åŒå€¼</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spin_singleValue">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radio_writeRandomValues">
          <property name="text">
           <string>å‘é€‰å®šè®¾å¤‡çš„æ‰€æœ‰å¯„å­˜å™¨å†™å…¥éšæœºå€¼</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radio_writeSpecificValues">
          <property name="text">
           <string>æŒ‰è¡¨æ ¼ä¸­çš„å¾…å†™å€¼å†™å…¥</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_selectDevices">
          <property name="text">
           <string>é€‰æ‹©è¦å†™å…¥çš„è®¾å¤‡ï¼š</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_updateDeviceList">
          <property name="text">
           <string>æ›´æ–°è®¾å¤‡åˆ—è¡¨</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_rightBottom">
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QListWidget" name="list_selectedDevices">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>80</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SelectionMode::MultiSelection</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
